ui/DesignSystem/Component/TxAccordion.svelte

```diff
@@ -0,0 +1,171 @@
+<script>
+  import { Flex, Icon, Text } from "../Primitive";
+  // TODO: Make component responsive to transactions change
+  export let transactions = null;
+  export let style = null;
+
+  const stateToColor = {
+    pending: "var(--color-caution)",
+    success: "var(--color-positive)",
+    error: "var(--color-negative)"
+  };
+
+  const stateToDescription = {
+    pending: "Pending",
+    success: "Success",
+    error: "Error"
+  };
+
+  const stateToIconState = {
+    pending: "caution",
+    success: "positive",
+    error: "negative"
+  };
+
+  const stateToSummary = {
+    pending: "pending",
+    success: "succeeded",
+    error: "failed"
+  };
+
+  const txSummary = {
+    pending: { count: 0, progress: 0 },
+    success: { count: 0, progress: 0 },
+    error: { count: 0, progress: 0 }
+  };
+
+  const fillSummary = () => {
+    transactions.forEach(tx => {
+      txSummary[tx.state].count += 1;
+      txSummary[tx.state].progress += tx.progress || 0;
+    });
+    txSummary["success"].progress = txSummary["success"].count * 100;
+  };
+
+  fillSummary();
+
+  const progressSummary =
+    txSummary["error"].count === transactions.length
+      ? null
+      : (txSummary["pending"].progress + txSummary["success"].progress) /
+        transactions.length;
+
+  const iconState =
+    txSummary["error"].count > 0
+      ? "negative"
+      : txSummary["pending"].count > 0
+      ? "caution"
+      : "positive";
+
+  const summaryState =
+    txSummary["pending"].count > 0
+      ? "pending"
+      : txSummary["error"].count > 0
+      ? "error"
+      : "success";
+
+  const summaryText = () => {
+    if (txSummary[summaryState].count > 1) {
+      return `${txSummary[summaryState].count} Transactions ${stateToSummary[summaryState]}`;
+    } else {
+      return `Transaction ${stateToSummary[summaryState]}`;
+    }
+  };
+
+  let hidden = true;
+
+  const hideCards = () => {
+    hidden = hidden ? false : true;
+  };
+</script>
+
+<style>
+  .pipeline {
+    border: 1px solid var(--color-foreground-level-3);
+    border-radius: 4px;
+    box-shadow: 0px 4px 8px var(--color-foreground-level-3-opacity-08);
+    width: 274px;
+  }
+
+  .negative {
+    border: 1px solid var(--color-negative);
+  }
+
+  .cards {
+    background-color: var(--color-foreground-level-1);
+    border-radius: 3px 3px 0 0;
+  }
+
+  .card {
+    border-bottom: 1px solid var(--color-foreground-level-3);
+    height: 64px;
+  }
+
+  .summary {
+    height: 56px;
+    background-color: var(--color-background);
+    border-radius: 3px;
+  }
+
+  .description {
+    align-self: center;
+    display: flex;
+    flex-direction: column;
+  }
+
+  .hidden {
+    display: none;
+  }
+</style>
+
+<div class="pipeline" class:negative={iconState === 'negative'} {style}>
+  <div class="cards" class:hidden>
+    {#each transactions as tx}
+      <!-- TODO: Link card to tx detail view -->
+      <div class="card">
```

Comment:
This nesting is quite expansive, can we refactor this part out into a distinct card component?

Reply:
In which way do you mean expensive?

I like the idea of having all the logic and styling of this component in one file. For now we also only need the `card` in this component so that I would tend to keeping it here. 
@rudolfs What's your opinion on this?

Reply:
Not expensive, wrote expansive, as in the nesting of the markup here is quite deep. It's not extraction for reusability but for readability. If there is no strong concern shared it's fine to leave it as is, this was the only place that gave me a stumble while reviewing.

Reply:
> @rudolfs What's your opinion on this?

I'm fine with it either way. I wish there was a better way of handling `<Flex>` without needing to nest it twice, but I don't know of a better solution at the moment either.